presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java

```diff
@@ -100,6 +106,22 @@ private static void verifyJavaVersion()
         }
     }
 
+    private static void verifyUsingG1Gc()
+    {
+        try {
+            ImmutableList<String> garbageCollectors = ManagementFactory.getGarbageCollectorMXBeans().stream()
+                    .map(GarbageCollectorMXBean::getName)
+                    .collect(toImmutableList());
+
+            if (garbageCollectors.stream().noneMatch(name -> name.toUpperCase(Locale.US).startsWith("G1 "))) {
+                warnRequirement("Current garbage collector are %s. Presto recommends the G1 garbage collector", garbageCollectors);
+            }
+        } catch (Exception e) {
```

Comment:
Can anything here throw checked exceptions? If not, this should be `RuntimeException`